lib/agent.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const { InvalidArgumentError, InvalidReturnValueError, NotSupportedError } = require('./core/errors')
+const { Pool } = require('../')
+const { weakCache, nodeMajorVersionIsGreaterThanOrEqualTo } = require('./core/util')
+
+const kAgentFactory = Symbol('agent factory')
+
+class Agent {
+  constructor (opts) {
+    /* istanbul ignore next */
+    if (!nodeMajorVersionIsGreaterThanOrEqualTo(14)) {
```

Comment:
I did some checks and `FinalizationRegistry` seems supported from version `14.6.0`. Wouldn't be better to check if `FinalizationRegistry` and `WeakRef` are defined globally? Assuming users might use version `14.5.0` or something like that.

Reply:
yeah good point - i saw how you did it in your PR and will copy that pattern